### PR TITLE
RHDEVDOCS-3283 Document advanced prunning options

### PR DESCRIPTION
**Pull request by sounix000**
- **Aligned team**: Dev Tools
- **OCP version for cherry-picking**: `enterprise-4.9`, `enterprise-4.10`
- **JIRA issues**: [RHDEVDOCS-3283 Document advanced prunning options](https://issues.redhat.com/browse/RHDEVDOCS-3283)
- **Preview pages**: [Automatic pruning of task run and pipeline run](https://deploy-preview-40748--osdocs.netlify.app/openshift-enterprise/latest/cicd/pipelines/automatic-pruning-taskrun-pipelinerun.html)
- **SME review**: @pradeepitm12     
- **Peer-review**: @rolfedh

### CODE REVIEW

File: cicd/pipelines/automatic-pruning-taskrun-pipelinerun.adoc [hunk 1]
```diff
@@ -0,0 +1,23 @@
+[id="automatic-pruning-taskrun-pipelinerun"]
+= Automatic pruning of task run and pipeline run 
+include::modules/common-attributes.adoc[]
+include::modules/pipelines-document-attributes.adoc[]
+:context: automatic-pruning-taskrun-pipelinerun
+
+toc::[]
+
+Old `TaskRun` and `PipelineRun` objects and their executed instances occupies physical resources that can be used for the active runs. To prevent this waste, {pipelines-title} provides annotations that cluster administrators can use to automatically prune the unused objects and their instances in various namespaces.
```
Comment (pradeepitm12) [inline cicd/pipelines/automatic-pruning-taskrun-pipelinerun.adoc:9]: How about `Old` --> `Stale`

File: modules/op-annotations-for-automatic-pruning-taskruns-pipelineruns.adoc [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+* `operator.tekton.dev/prune.strategy`: Set the value of this annotation to either `keep` or `keep-since`.
+
+As an example, consider the following annotations that retain all task runs and pipeline runs created in the last 5 days, and deletes the older resources: 
+
+.Example of auto-pruning annotations
+[source,yaml]
+----
+...
+  annotations:
```
Comment (pradeepitm12) [inline modules/op-annotations-for-automatic-pruning-taskruns-pipelineruns.adoc:33]: :+1:

File: cicd/pipelines/automatic-pruning-taskrun-pipelinerun.adoc [hunk 3]
```diff
@@ -0,0 +1,22 @@
+[id="automatic-pruning-taskrun-pipelinerun"]
```
Comment (rolfedh) [inline cicd/pipelines/automatic-pruning-taskrun-pipelinerun.adoc:1]: Would you apply the new Jupiter apply to this document?
https://docs.engineering.redhat.com/pages/viewpage.action?spaceKey=JUP&title=Jupiter+Source+files+readiness+Checklist

File: cicd/pipelines/automatic-pruning-taskrun-pipelinerun.adoc [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+Stale `TaskRun` and `PipelineRun` objects and their executed instances occupy physical resources that can be used for the active runs. To prevent this waste, {pipelines-title} provides annotations that cluster administrators can use to automatically prune the unused objects and their instances in various namespaces. 
+
+[NOTE]
+====
+* Starting with {pipelines-title} 1.6, auto-pruning is enabled by default.
+
+* Configuring automatic pruning by specifying annotations affects the entire namespace. You cannot selectively auto-prune individual task runs and pipeline runs in a namespace.
+====
+
+include::modules/op-annotations-for-automatic-pruning-taskruns-pipelineruns.adoc[leveloffset=+2]
```
Comment (rolfedh) [inline cicd/pipelines/automatic-pruning-taskrun-pipelinerun.adoc:17]:
```suggestion
include::modules/op-annotations-for-automatic-pruning-taskruns-pipelineruns.adoc[leveloffset=+1]
```

File: modules/op-annotations-for-automatic-pruning-taskruns-pipelineruns.adoc [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+// cicd/pipelines/automatic-pruning-taskrun-pipelinerun.adoc
+
+[id="annotations-for-automatic-pruning-taskruns-pipelineruns_{context}"]
+= Annotations for automatically pruning task runs and pipeline runs 
+
+To automatically prune task runs and pipeline runs in a namespace, you can set the following annotations:
+
+* `operator.tekton.dev/prune.schedule`: If the value of this annotation is different from the value specified in the `TektonConfig` custom resource definition, a new cron job in that namespace is created.
+
+* `operator.tekton.dev/prune.skip`: When set to `true`, the namespace for which it is configured will not be pruned.
```
Comment (rolfedh) [inline modules/op-annotations-for-automatic-pruning-taskruns-pipelineruns.adoc:12]:
```suggestion
* `operator.tekton.dev/prune.skip`: When set to `true`, the namespace for which it is configured is not pruned.
```

File: modules/op-annotations-for-automatic-pruning-taskruns-pipelineruns.adoc [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+* `operator.tekton.dev/prune.resources`: This annotation accepts a comma-separated list of resources. To prune a single resource such as a pipeline run, set this annotation to `"pipelinerun"`. To prune multiple resources, such as task run and pipeline run, set this annotation to `"taskrun, pipelinerun"`.
```
Comment (rolfedh) [inline modules/op-annotations-for-automatic-pruning-taskruns-pipelineruns.adoc:14]: You want the user to enter the annotation between double-quotes, correct?

File: modules/op-annotations-for-automatic-pruning-taskruns-pipelineruns.adoc [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+* `operator.tekton.dev/prune.keep-since`: Use this annotation to retain resources based on their age. The value for this annotation must be equal to the age of the resource in minutes. For example, to retain resources which were created not more than five days ago, set `keep-since` to `7200`.
++
+[NOTE]
+====
+The `keep` and `keep-since` annotations are mutually exclusive. For any resource, you must configure only one of them.
+====
+
+* `operator.tekton.dev/prune.strategy`: Set the value of this annotation to either `keep` or `keep-since`.
+
+As an example, consider the following annotations that retain all task runs and pipeline runs created in the last 5 days, and deletes the older resources:
```
Comment (rolfedh) [inline modules/op-annotations-for-automatic-pruning-taskruns-pipelineruns.adoc:27]:
```suggestion
For example, consider the following annotations that retain all task runs and pipeline runs created in the last 5 days, and deletes the older resources: 
```

**Review by rolfedh — COMMENTED** [review]
Excellent work, Souvik. I left some suggestions and highlighted the new requirements for Jupiter.

File: modules/op-annotations-for-automatic-pruning-taskruns-pipelineruns.adoc (hunk 6 above)
Comment (sounix000) [reply modules/op-annotations-for-automatic-pruning-taskruns-pipelineruns.adoc:14]: Yes, it's treated as a string as a whole.

**Review by rolfedh — APPROVED** [review]

**Pull request merged**